apps/api/src/app/environments/usecases/generate-unique-api-key/generate-unique-api-key.usecase.ts

```diff
@@ -27,6 +27,18 @@ export class GenerateUniqueApiKey {
     return apiKey as string;
   }
 
+  private async validateIsApiKeyUsed(apiKey: string) {
+    const encryptedApiKey = encryptApiKey(apiKey);
+
+    // backward compatibility - update to `findByApiKey` after encrypt-api-keys-migration run
+    const environment = await this.environmentRepository.findByApiKeyBackwardCompatibility({
+      decryptedKey: apiKey,
+      encryptedKey: encryptedApiKey,
+    });
```

Comment:
let's have a ticket for this so we won't forget

Reply:
Good point! i will create one.